mlrun/serving/states.py

```diff
@@ -1508,15 +1514,31 @@ def _init_async_objects(context, steps):
                     name=step.name,
                     context=context,
                 )
-            if not step.next and hasattr(step, "responder") and step.responder:
+            if (
+                respond_supported
+                and not step.next
+                and hasattr(step, "responder")
+                and step.responder
+            ):
                 # if responder step (return result), add Complete()
                 step.async_object.to(storey.Complete(full_event=True))
                 wait_for_result = True
 
     source_args = context.get_param("source_args", {})
+
+    # list from https://github.com/nuclio/nuclio/blob/1.12.0/pkg/platform/abstract/platform.go#L1546
+    explicit_ack_supported = trigger in [
+        "v3io-stream",
+        "v3ioStream",
+        "kafka-cluster",
+        "kafka",
+    ]
+    # only turn explicit ack on if trigger supports explicit ack
+    explicit_ack = explicit_ack_supported and mlrun.mlconf.is_explicit_ack()
+
     default_source = storey.SyncEmitSource(
         context=context,
-        explicit_ack=mlrun.mlconf.is_explicit_ack(),
+        explicit_ack=explicit_ack,
```

Comment:
do you need to do the same also here? https://github.com/mlrun/mlrun/blob/3f68331fa7c2dcd339165f8c3522a39fbc51bf2b/mlrun/datastore/sources.py#L796